krainboltgreene/library.io
# Scaffold for new libraries

This gives the binary a sub-command for building new libraries based on library.io

Here's an example usage for the command:

``` shell
$ library new shogun -b
Creating a new library named "shogun.io" in your current directory...

  Creating shogun.io/                       [dir]
  Creating shogun.io/test/                  [dir]
  Creating shogun.io/test/shogun.io         [file]
  Creating shogun.io/test/shogun/           [dir]
  Creating shogun.io/test/shogun/sample.io  [file]
  Creating shogun.io/lib/                   [dir]
  Creating shogun.io/lib/shogun.io          [file]
  Creating shogun.io/lib/shogun/            [dir]
  Creating shogun.io/lib/shogun/sample.io   [file]
  Creating shogun.io/ex/                    [dir]
  Creating shogun.io/ex/ex1.io              [file]
  Creating shogun.io/bin/                   [dir]
  Creating shogun.io/bin/shogun.io          [file]

Done! To start working on this library:

  * Change into the directory [`cd shogun.io`]
  * Make sure the sample tests run [`library test`]
  * Start writing tests
  * Write code to make the tests run
  * Refact your code!
```

Acceptance Criteria:
- Developer should be able to invoke the binary with the new sub-command
- If invoking the command naked (`library new`) it should show a help banner
- There should be a man file for the binary (and the sub-command)
- It should create a root directory with the directory name based on the project name + ".io"
- Root dir should be made in the current directory or accept a relative directory
- It should create a `lib`, `test`,  and `ex` directory
- It should create a `README.mdown`, `LICENSE`, and `lib.io` file in the root directory
- It should create a `[libname].io` file and `[libname]/` directory inside the `lib/` directory
- It should create a `[libname].io` file and `[libname]/` directory inside the `test/` directory
- It should create a `ex1.io` file inside the `ex/` directory
- If the `-b` or `--bin` flag is provided then it should create a `bin/` directory in the root directory
- If the `-b` or `--bin` flag is provided then it should create a `[libname]` file in the root directory, with execute permissions
- It should create a `sample.io` file in the `test/[libname]/` and `lib/[libname]/` directories
